### PR TITLE
Mocha/Chai for JavaScript testing

### PR DESCRIPTION
**Pull request by allenmoore**
The pull request adds both Mocha.js and Chai.js for JavaScript testing. Developers have two options to run a mocha test:
1. Run the grunt cli commands: `grunt mocha` or `grunt test`.
2. Run in browser at `local.dev/path/to/tests/mocha/file-name.html`.

### CODE REVIEW

**Comment by lkwdwrd** [comment]
Looks pretty good!

One suggestion -- if we go with Mocha, I'd rather just cut over to it completely and pull out qunit, or is there do you feel there is a good reason not to do that?

**Comment by allenmoore** [comment]
Hey @lkwdwrd,

If we go with Mocha, I would completely pull out QUnit. Since both are serving the same purpose, it would seem pointless to have to maintain both. Also, Mocha is specified in the 10up Engineering Best Practices as what we use for testing -- https://10up.github.io/Engineering-Best-Practices/javascript/#unit-and-integration-testing, so it would seem fitting for it be a part of the generator. What are your thoughts?

**Comment by lkwdwrd** [comment]
Completely agree, which is why I was asking. Still saw a bunch of qunit in your PR :-)

**Comment by allenmoore** [comment]
What? I don't see QUnit anywhere...now that I've completely removed it. ;)

File: shared/grunt/tasks/options/_mocha.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+var mochaPath = 'tests/mocha/';
+
+module.exports = {
+	test: {
+		src: [ mochaPath + '**/*.php', mochaPath + '**/*.html' ],
```
Comment (ericmann) [inline shared/grunt/tasks/options/_mocha.js]: Um ... are we sure we want Mocha testing php files?

**Comment by allenmoore** [comment]
@ericmann Yeah, that could present some potential issues. PHP files removed.

**Pull request merged**